src/NovelRT/Windowing/WindowingService.cpp

```diff
@@ -4,42 +4,80 @@
 
 namespace NovelRT::Windowing {
   WindowingService::WindowingService(NovelRunner* const runner) :
-    _window(std::unique_ptr<SDL_Window, decltype(&SDL_DestroyWindow)>(nullptr, SDL_DestroyWindow)),
+    _window(std::unique_ptr<GLFWwindow, decltype(&glfwDestroyWindow)>(nullptr, glfwDestroyWindow)),
     _logger(LoggingService(Utilities::Misc::CONSOLE_LOG_WINDOWING)),
-    _runner(runner) {
-    runner->getInteractionService()->subscribeToResizeInputDetected([this](auto value) {
-      _windowSize = value;
-      _logger.logInfo("New size detected! Notifying GFX and other members...");
-      raiseWindowResized(_windowSize); });
+    _runner(runner), _isTornDown(false) {}
+
+  void WindowingService::errorCallback(int, const char* error) {
+    _logger.logError("Could not initialize GLFW: ", error);
   }
 
   void WindowingService::initialiseWindow(int displayNumber, const std::string& windowTitle) {
-    if (SDL_Init(SDL_INIT_VIDEO | SDL_INIT_EVENTS) < 0) {
-      _logger.logError("Could not initialize SDL2: ", std::string(SDL_GetError()));
-      throw std::runtime_error("Unable to continue! SDL2 failed to initialise.");
-    }
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 0);
+    glfwWindowHint(GLFW_CLIENT_API, GLFW_OPENGL_ES_API);
+    glfwWindowHint(GLFW_AUTO_ICONIFY, GLFW_FALSE);
 
-    SDL_DisplayMode displayData;
-    SDL_GetCurrentDisplayMode(displayNumber, &displayData);
+    GLFWmonitor* primaryMonitor = glfwGetPrimaryMonitor();
+    const GLFWvidmode* displayData = glfwGetVideoMode(primaryMonitor);
 
     // create window
-    float wData = displayData.w * 0.7f;
-    float hData = displayData.h * 0.7f;
-    auto window = SDL_CreateWindow(
-      windowTitle.c_str(), SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
-      wData, hData, SDL_WINDOW_OPENGL | SDL_WINDOW_SHOWN | SDL_WINDOW_RESIZABLE);
-
-    _window = std::unique_ptr<SDL_Window, decltype(&SDL_DestroyWindow)>(window, SDL_DestroyWindow);
-    _windowSize = Maths::GeoVector<float>(wData, hData);
+    float wData = displayData->width * 0.7f;
+    float hData = displayData->height * 0.7f;
+    auto window = glfwCreateWindow(wData, hData, _windowTitle.c_str(), NULL, NULL);
```

Comment:
Please fix this to not use C-style `NULL`, it looks out of place

Reply:
The NovelRT solution was checked for `NULL`s and changed _most_ to `nullptr`; `ALuint` seems to be incompatible with `nullptr` as per VS (causes build failure) sooooo they should be the only ones left.